drivers/ethernet/eth_mcux.c

```diff
@@ -166,6 +177,64 @@ rx_buffer[CONFIG_ETH_MCUX_RX_BUFFERS][ETH_MCUX_BUFFER_SIZE];
 static u8_t __aligned(ENET_BUFF_ALIGNMENT)
 tx_buffer[CONFIG_ETH_MCUX_TX_BUFFERS][ETH_MCUX_BUFFER_SIZE];
 
+#ifdef CONFIG_NET_POWER_MANAGEMENT
+static void eth_mcux_phy_enter_reset(struct eth_context *context);
+void eth_mcux_phy_stop(struct eth_context *context);
+
+static int eth_mcux_device_pm_control(struct device *dev, u32_t command,
+				      void *context, device_pm_cb cb, void *arg)
+{
+	struct eth_context *eth_ctx = (struct eth_context *)dev->driver_data;
+	int ret = 0;
+
+	if (!eth_ctx->clock_dev) {
+		LOG_ERR("No CLOCK dev");
+
+		ret = -EIO;
+		goto out;
+	}
+
+	if (command == DEVICE_PM_SET_POWER_STATE) {
+		if (*(u32_t *)context == DEVICE_PM_SUSPEND_STATE) {
+			LOG_DBG("Suspending");
+
+			ret = net_if_suspend(eth_ctx->iface);
+			if (ret == -EBUSY) {
+				goto out;
+			}
+
+			eth_mcux_phy_enter_reset(eth_ctx);
+			eth_mcux_phy_stop(eth_ctx);
+
+			ENET_Deinit(eth_ctx->base);
+			clock_control_off(eth_ctx->clock_dev,
+				(clock_control_subsys_t)eth_ctx->clock);
```

Comment:
First, add a ENET_Reset() before the Deinit(). I think that will block any transient events that might generate an interrupt.

Second, you don't need the clock_control_off because the ENET_Deinit() already does this. Now @MaureenHelm  and I have been discussing this because one could argue that maybe the NXP HAL shouldn't do that and there are compile options to avoid this but that might have ramifications in other areas.

Reply:
Well, I guess atm it does not harm to let the clock_control stuff (it's redundant at worse case).
I'll update the ethernet driver.